crypto/hasher/hasher_test.go

```diff
@@ -0,0 +1,55 @@
+package hasher
+
+import (
+	"encoding/hex"
+	"testing"
+)
+
+// h2h converts a hex string into its Hash object.
+func h2h(h string) Hash {
+	b, err := hex.DecodeString(h)
+	if err != nil {
+		panic("invalid hex string")
+	}
+	var ret Hash
+	copy(ret[:], b)
+	return ret
+}
+
+// s2h converts a byte slice into its Hash object.
+func s2h(s []byte) Hash {
+	var ret Hash
+	copy(ret[:], s)
+	return ret
+}
+
+func TestHashLeafVectors(t *testing.T) {
+	for _, tc := range []struct {
+		treeNonce [32]byte // treeNonce is treeID in KT, it should be a 32-byte array for cross-project compatibility
+		index     []byte
+		depth     uint32
+		leaf      []byte
+		want      Hash
+	}{
+		{treeNonce: [32]byte{0}, index: []byte("foo"), depth: 128, leaf: []byte("leaf"), want: h2h("65e7f29787a6168affd016656bb1f4f03af91cf7416270f5015005f8594d3eb6")},
```

Comment:
Question about the index since I ran into this too. 
Should indexes be required to be the full length of the hash function? This would change the test vectors.